202112_timeseries/time-series-forecasting.Rmd

```diff
@@ -0,0 +1,699 @@
+
+
+```{r}
+mod_linear
+
+## you can look at model statistics 
+summary(mod_linear)
+
+plot(mod_linear)
+```
+
+
+### Further notes on the use of simulated data:
+
+Please pardon the digression from time series into the power of simulation for power analysis.
+
+The data generated above has a random error term $\epsilon\sim N(0,1)$. You can use simulation for power analysis by simulating a lot of datasets and fitting the model to test, e.g., what percent of the datasets have a particular P value, or if the mean parameter estimates (intercept, slope, error in this case) converge on the 'true' values.
+
+The concept of a 'data generating process' is an important concept when modeling, and when synthesizing data from many different locations. A 'data generating process' includes both the system being studied and the tools used to observe (discussed further below in the section on state space models). It is common to hear data referred to as the truth. But data can only represent an incomplete view of the system itself.  In the end both data and models are representations of a system that help estimate some underlying truth.
+
+```{r}
+
+n_sims <- 1000
+sim_data <- list()
+fits <- list()
+days <- 0:9
+n <- length(days)
+
+
+for(i in 1:n_sims){
+  sim_data[[i]] <- vector(mode = 'double', length = n)
+  for(t in seq_along(days)){
+    mass[t] <- 1 + 2 * days[t] + rnorm(1, 0, 1)
+  }
+  sim_data[[i]] <- mass
+  fits[[i]] <- lm(mass ~ 1 + days)
+}
+```
+
+
+
+Now, we can create a data frame of the statistics that we are interested in. Here, we will look at fitted values of:
+
+1. intercept
+2. slope
+3. model error
+4. significance (P value) of test that intercept is different than 0
+5. significance of test that slope is different than 0
+
+
+```{r}
+v <- vector(length = n_sims)
+stats <- data.frame(intercept = v, slope = v, sigma = v, p_i = v, p_s = v)
+for(i in 1:n_sims){
+  stats[i, c('intercept', 'slope')] <- coef(fits[[i]])
+  stats[i, 'sigma']                 <- sigma(fits[[i]])
+  stats[i, c('p_i', 'p_s')]         <- summary(fits[[i]])$coefficients[1:2, 4]
+}
+
+
+```
+
+Then, we can check out whether our model returns the simulated terms. Here we will look at the mean and sd, and create a nice table while we are at it! 
+
+```{r}
+data.frame(slope = c(mean(stats$slope), sd(stats$slope)), 
+           intercept = c(mean(stats$intercept), sd(stats$intercept)), 
+           row.names = c('mean', 'sd'))
+
+```
+
+Now that we are working with our 'population' of model fits, we can use all manner of statistics and visualizations to better understand the simulated experiments. We can use `pairs`, `hist`, `quantile` and others. 
+
+For the classical 'power analysis' lets look at the percent of time the slope term is significant at 
+P <= 0.05. How often is the slope significant? How often is the intercept significant?
+
+This is rather straightforward with the 'stats' data.frame we created above:
+
+```{r}
+
+sum(intercept_ps > 0.05)/length(day_ps)
```

Comment:
intercept_ps and day_ps don't exist at this point

Reply:
`sum(stats$p_i > 0.05)/nrow(stats)`
`sum(stats$p_s > 0.05)/nrow(stats)`
Is this what was intended?